### PR TITLE
Deprecate state=list in ec2_vol

### PR DESCRIPTION
**Pull request by jillr**
##### SUMMARY
Deprecate state=list in ec2_vol
Functionality is already available in ec2_vol_info
Add missing return docs to ec2_vol_info

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
 ec2_vol
 ec2_vol_info

### CODE REVIEW

**Comment by jillr** [comment]
cc @tremble

File: plugins/modules/ec2_vol_info.py [hunk 1]
```diff
@@ -51,9 +51,66 @@
...
+            type: dict
+            sample: {
+                env: "dev"
+                }
+        type:
+            description: The volume type. This can be gp2, io1, st1, sc1, or standard.
+            type: str
+            sample: "standard"
+        zone:
+         description:
```
Comment (tremble) [inline plugins/modules/ec2_vol_info.py:110]:
```suggestion
          description: The Availability Zone of the volume
```

Not 100% on the whitespace here it's difficult to compare the 'real' and suggested indentaion.

Comment (jillr) [reply plugins/modules/ec2_vol_info.py:110]: White is totally wrong, thanks.
I think you did the deprecation for ec2_tag. I didn't find any others at least

File: plugins/modules/ec2_vol_info.py [hunk 2]
```diff
@@ -51,9 +51,66 @@
...
+            type: dict
+            sample: {
+                env: "dev"
+                }
+        type:
+            description: The volume type. This can be gp2, io1, st1, sc1, or standard.
+            type: str
+            sample: "standard"
+        zone:
+            description:
```
Comment (tremble) [inline plugins/modules/ec2_vol_info.py:110]:
```suggestion
            description: The Availability Zone of the volume
```

It feels wrong to leave the description here blank when we do know what it is.

**Comment by jillr** [comment]
they would fail if I didn't bother to make the changelog valid yaml  :)

**Pull request merged**